libsolidity/parsing/Parser.cpp

```diff
@@ -105,6 +107,20 @@ ASTPointer<SourceUnit> Parser::parse(shared_ptr<Scanner> const& _scanner)
 	}
 }
 
+void Parser::parsePragmaVersion(vector<Token> tokens, vector<string> literals)
+{
+	SemVerMatchExpressionParser parser(tokens, literals);
+	auto matchExpression = parser.parse();
+	SemVerVersion currentVersion{string(VersionString)};
+	// FIXME: only match for major version incompatibility
+	if (!matchExpression.matches(currentVersion))
+		fatalParserError(
```

Comment:
Why fatal?

Reply:
Because the main point is that we do not fail with random parser errors afterwards.

Reply:
But it will only report the error in the first source it finds. We can do it like that, just wanted to ask.

Reply:
Not entirely sure what is better, but have a feeling that other parsing errors might become fairly confusing and then this one is overlooked by users. From a practical view though if it is an unknown version, that means we cannot parse it and a "fatal" error seems more fitting.

Reply:
ok!